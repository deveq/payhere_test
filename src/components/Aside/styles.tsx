import styled from 'styled-components';
import {flexCenter} from 'styles/helper';

export const Aside = styled.aside`
    width: 200px;
    background: ${({theme}) => theme.bg_page1};
    padding: 30px;
    box-sizing: border-box;
    flex-shrink: 0;
`;

export const List = styled.ul`
    display: grid;
    gap: 5px;
`;

export const Title = styled.h2`
    font-size: 24px;
    font-weight: bold;
    border-bottom: ${({theme}) => `1px solid ${theme.bg_element4}`};
    color: ${({theme}) => theme.text1};
    padding-bottom: 10px;
    margin-bottom: 20px;
    text-indent: 10px;
`;

export const ListItem = styled.li<{selected: boolean}>`
    background: ${({selected,theme}) => selected ? theme.bg_element1 : 'transparent'};
    padding: 15px 20px;
    border-radius: 10px;
    font-weight: bold;
    color: ${({selected, theme}) => selected ? theme.text2 : theme.text3};
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element1};
    };
    cursor: pointer;
`;

export const Subtitle = styled.div`

`;