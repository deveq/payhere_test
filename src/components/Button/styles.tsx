import styled from "styled-components";

type ButtonSize = 'sm' | 'md' | 'lg';
export interface ButtonProps {
    size?: ButtonSize;
    backgroundColor?: string;
    borderColor?: string;
    color?: string;
    showShadow?: boolean;
}

const handleSize = (size?: ButtonSize) => {
    if (size === 'sm') return '0.5rem 1rem';
    if (size === 'md') return '0.75rem 1.5rem';
    if (size === 'lg') return '1rem 2rem';
    return '0.75rem 1.5rem';
}

export const Button = styled.button<ButtonProps>`
    background: ${({backgroundColor, theme}) => 
        backgroundColor || theme.bg_element2
    };
    border-radius: 5px;
    color: ${({color, theme}) => 
        color || theme.text1
    };
    padding: ${({size}) => handleSize(size)};
    font-weight: bold;
    box-shadow: ${({theme, showShadow}) => showShadow ? theme.boxShadow : 'none'};
    border: ${({showShadow, theme}) => showShadow ? 'none' : `1px solid ${theme.border2}` };
    &:hover,
    &:active {
        background: ${({theme}) => theme.bg_element3};
    }
    cursor: pointer;
`;

Button.defaultProps = {
    size: 'md',
}

export const Primary = styled(Button)`
    background: ${({theme}) => theme.primary};
    color: white;
`;

export const Destructive = styled(Button)`
    background: ${({theme}) => theme.destructive};
    color: white;
    border: none;
    &:hover,
    &:active {
        background: ${({theme}) => theme.destructive};
    }
`;