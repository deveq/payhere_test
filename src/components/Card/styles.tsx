import styled from 'styled-components';

export const CardContainer = styled.div`
    background: ${({theme}) => theme.bg_element1};
    border-radius: 10px;
    padding: 5px 10px;
`;