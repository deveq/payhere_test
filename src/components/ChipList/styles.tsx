import styled from 'styled-components';

interface ChipProps {
    selected?: boolean;
}

export const Chip = styled.div<ChipProps>`
    padding: 5px 10px;
    max-width: 150px;
    min-width: 40px;
    text-align: center;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    border-radius: 20px;
    border: 1px solid ${({theme}) => theme.border2};
    background: ${({selected, theme}) => selected ? theme.primary : 'white'};
    color: ${({selected}) => selected ? 'white' : '#333'};
    cursor: pointer;
    &:active {
        opacity: 0.9;
    }
`;

export const ChipList = styled.div`
    display: flex;
    /* flex-direction: column;
    align-items: flex-start; */
    flex-wrap: wrap;
    gap: 8px;
    margin-bottom: 10px;
`;

export const Message = styled.p`
    color: ${({theme}) => theme.text1};
`;