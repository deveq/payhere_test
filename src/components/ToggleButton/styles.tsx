import styled from 'styled-components';

interface SpaceProps {
    clicked?: boolean;
}

export const Container = styled.div<SpaceProps>`
    width: 60px;
    height: 30px;
    background-color: ${({clicked}) => clicked ? 'green' : 'gray'};
    display: flex;
    border-radius: 30px;
    transition: all 0.5s ease;
`;

export const Thumb = styled.div<SpaceProps>`
    height: 30px;
    width: 30px;
    border-radius: 15px;
    background-color: ${({clicked}) => clicked ? 'white' : 'whitesmoke'};
    border: 2px solid ${({clicked}) => clicked ? 'green' : 'gray'};
    box-sizing: border-box;
    transition: all 0.5s ease;

`;

export const Space = styled.div<SpaceProps>`
    flex: ${({clicked}) => clicked ? 1 : 0};
    transition: all 0.5s ease;
`;

