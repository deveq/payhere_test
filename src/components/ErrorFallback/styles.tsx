import styled from 'styled-components';
import {flexCenter} from 'styles/helper';
import {FaRegSadTear} from 'react-icons/fa';

export const Container = styled.div`
    height: 100vh;
    ${flexCenter};
    flex-direction: column;
`;

export const Paragraph = styled.p`
    font-size: 30px;
`;

export const ErrorMessage = styled.pre`
    font-size: 20px;
`;

export const SadIcon = styled(FaRegSadTear)`
    width: 100px;
    height: 100px;
    color: gray;
`;