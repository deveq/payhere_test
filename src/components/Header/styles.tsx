import React from 'react';
import styled from 'styled-components';
import { ReactComponent } from 'assets/logo.svg';
import {flexCenter} from 'styles/helper';


export const Container = styled.header`
	display: flex;
	padding: 1rem 2rem;
	background: ${(props) => props.theme.bg_page1};
	box-shadow: ${(props) => props.theme.boxShadow};
`;

export const Logo = styled(ReactComponent)`
	width: 50px;
	height: 50px;
	${flexCenter};
`;

export const Headline = styled.h1`
`;

export const Nav = styled.nav`
	flex: 1;
	${flexCenter};
`;

export const LoginContainer = styled.div`

`;