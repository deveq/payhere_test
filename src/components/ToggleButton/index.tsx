import * as Styled from './styles';

interface ToggleButtonProps {
    clicked: boolean;
    onClick: () => void;
}

const ToggleButton = ({clicked, onClick}: ToggleButtonProps) => {
    return (
        <Styled.Container onClick={onClick} clicked={clicked}>
            <Styled.Space clicked={clicked} />
            <Styled.Thumb clicked={clicked} />
        </Styled.Container>
    )
}

export default ToggleButton;