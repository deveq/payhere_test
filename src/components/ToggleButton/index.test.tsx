import {screen} from '@testing-library/react';
import {renderWithWrapper} from 'tests';
import {lightTheme} from 'styles/theme';
import ToggleButton from './index';
import 'jest-styled-components';

describe('<ToggleButton />', () => {
    it('정상적으로 렌더링 되었는지 확인한다', () => {
        const fn = jest.fn();
        const clicked = false;
        const {container} = renderWithWrapper(<ToggleButton 
            onClick={fn}
            clicked={clicked}
        />);

        const toggleButton = screen.getByTitle('toggleButton');
        expect(toggleButton).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    })

    it('props에 따라 정상적으로 렌더링 되었는지 확인한다', () => {
        const fn1 = jest.fn();
        const fn2 = jest.fn();
        renderWithWrapper(
        <>
        <ToggleButton 
            onClick={fn1}
            clicked={false}
        />
        <ToggleButton 
            onClick={fn2}
            clicked={true}
        />
        </>
        );

        const [falseButton, trueButton] = screen.getAllByTitle('toggleButton');
        expect(falseButton).toBeInTheDocument();
        expect(trueButton).toBeInTheDocument();
        expect(falseButton).toHaveStyleRule(
            'background',
            'gray'
        );
        expect(trueButton).toHaveStyleRule(
            'background',
            '#05c3a7'
        );

        expect(falseButton.children[1]).toHaveStyle({
            background: 'whitesmoke',
            border: '2px solid gray',
        });

        expect(trueButton.children[1]).toHaveStyle({
            background: 'white',
            border: '2px solid #05c3a7',
        });

        expect(falseButton.children[0]).toHaveStyle({
            flex: 0
        });

        expect(trueButton.children[0]).toHaveStyle({
            flex: 1,
        });

    })
})