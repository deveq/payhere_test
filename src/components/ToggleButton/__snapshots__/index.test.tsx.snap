// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton /> 정상적으로 렌더링 되었는지 확인한다 1`] = `
.c0 {
  width: 60px;
  height: 30px;
  background: gray;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 30px;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
  cursor: pointer;
}

.c2 {
  height: 30px;
  width: 30px;
  border-radius: 15px;
  background: whitesmoke;
  border: 2px solid gray;
  box-sizing: border-box;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

.c1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  -webkit-transition: all 0.5s ease;
  transition: all 0.5s ease;
}

<div>
  <div
    class="c0"
    title="toggleButton"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    />
  </div>
</div>
`;
